### PR TITLE
feat: optimize validation

### PR DESCRIPTION
**Pull request by hwbrzzl**
## 📑 Description

SafeData only contains the data that is defined in the rules, we want user can the original data that is not defined in the rules, so that user doesn't need to define rules for all fields.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for converting structs to maps, enhancing dynamic data handling.
- **Improvements**
	- Enhanced spinner color representation for better visual output across terminal environments.
	- Improved data binding logic for greater flexibility when merging various data types and preserving existing data.
- **Refactor**
	- Renamed and reorganized test cases for improved clarity and maintainability, resulting in better validation coverage.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

File: console/cli_context.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func (r *CliContext) Secret(question string, option ...console.SecretOption) (st
 }
 
 func (r *CliContext) Spinner(message string, option console.SpinnerOption) error {
-	style := lipgloss.NewStyle().Foreground(lipgloss.Color("#8ED3F9"))
+	style := lipgloss.NewStyle().Foreground(lipgloss.CompleteColor{TrueColor: "#3D8C8D", ANSI256: "30", ANSI: "6"})
```
Comment (hwbrzzl) [inline console/cli_context.go:233]: As the description: https://github.com/charmbracelet/lipgloss?tab=readme-ov-file#true-color-16777216-colors-24-bit, the color will be automatically detected by Terminal, it's unexpected. We should use the CompleteColor.

File: validation/validator.go [hunk 2]
```diff
@@ -38,21 +38,39 @@ func (v *Validator) Bind(ptr any) error {
 		return nil
 	}
 
+	// SafeData only contains the data that defined in the rules,
+	// we want user can get the original data that not defined in the rules,
+	// so that user doesn't need to define rules for all fields.
```
Comment (hwbrzzl) [inline validation/validator.go:43]: Notice

**Comment by hwbrzzl** [comment]
@coderabbitai resolve

File: support/maps/maps.go [hunk 3]
```diff
@@ -25,6 +29,54 @@ func Forget[K comparable, V any](mp map[K]V, keys ...K) {
...
+		}
+
+		if fieldValue.Kind() == reflect.Pointer {
+			if fieldValue.IsNil() {
+				res[fieldType.Name] = nil
+				continue
+			}
+
+			fieldValue = fieldValue.Elem()
+		}
```
Comment (devhaozi) [inline support/maps/maps.go:61]: Can be optimize to `fieldValue = reflect.Indirect(fieldValue)`

Comment (hwbrzzl) [reply support/maps/maps.go:61]: If so, how to add:

```
if fieldValue.IsNil() {
  res[fieldType.Name] = nil
  continue
}
```

**Review by devhaozi — APPROVED** [review]
LGTM

**Pull request merged**